### PR TITLE
Run pyupgrade - V

### PR DESCRIPTION
**Pull request by iamsobanjaved**
JIRA: https://openedx.atlassian.net/browse/BOM-1978

### CODE REVIEW

**Review by UsamaSadiq — APPROVED** [review]

File: course_discovery/apps/course_metadata/tests/test_models.py [hunk 1]
```diff
@@ -1359,14 +1358,14 @@ def test_search(self):
         Verify that the program endpoint correctly handles basic elasticsearch queries
         """
         query = 'title:' + self.program.title
-        self.assertSetEqual(set(Program.search(query)), set([self.program]))
+        self.assertSetEqual({Program.search(query).first()}, {self.program})
```
Comment (Dillon-Dumesnil) [inline course_discovery/apps/course_metadata/tests/test_models.py:1390]: Why did you need the `.first()` here, but not below in the `test_subject_search` method?

File: course_discovery/apps/course_metadata/tests/test_models.py [hunk 2]
```diff
@@ -1326,11 +1324,12 @@ def tearDown(self):
         """
         Resets course canonical_course_runs to initial state.
         """
-        super(ProgramTests, self).tearDown()
-        for course_run in self.course_runs[:2]:
-            course = course_run.course
-            course.canonical_course_run = None
-            course.save()
+        super().tearDown()
+        with transaction.atomic():
```
Comment (Dillon-Dumesnil) [inline course_discovery/apps/course_metadata/tests/test_models.py]: Was this necessary or just something to add in to ensure cleanliness?

**Review by Dillon-Dumesnil — APPROVED** [review]

File: course_discovery/apps/course_metadata/utils.py [hunk 3]
```diff
@@ -657,7 +657,7 @@ def handle_tag(self, tag, attrs, start):
             if attrs:
                 attr_dict = dict(attrs)
                 if start and 'lang' in attr_dict:
-                    self.outtextf(u'<span lang="{}">'.format(attr_dict['lang']))
+                    self.outtextf('<span lang="{}">'.format(attr_dict['lang']))
```
Comment (ciduarte) [inline course_discovery/apps/course_metadata/utils.py]: Should this be `f'<span lang="{attr_dict['lang']}">'`?

Comment (iamsobanjaved) [reply course_discovery/apps/course_metadata/utils.py]: `pyupgrade` didn't change this. But if we do, it will be `f"<span lang=\"{attr_dict['lang']}\">"`, is this okay?

Comment (ciduarte) [reply course_discovery/apps/course_metadata/utils.py]: Could this work to avoid the `\"`?
```
lang = attr_dict['lang']
self.outtextf(f'<span lang="{lang}">')
```

Comment (iamsobanjaved) [reply course_discovery/apps/course_metadata/utils.py]: Added that

**Review by awais786 — APPROVED** [review]
